core/devmode/src/main/java/io/quarkus/dev/DevModeMain.java

```diff
@@ -43,24 +29,16 @@
     public static final String DEV_MODE_CONTEXT = "META-INF/dev-mode-context.dat";
     private static final Logger log = Logger.getLogger(DevModeMain.class);
 
-    private static volatile ClassLoader currentAppClassLoader;
-    private static volatile URLClassLoader runtimeCl;
     private final DevModeContext context;
 
-    private static volatile Closeable runner;
-    static volatile Throwable deploymentProblem;
-    static volatile Throwable compileProblem;
-    static volatile RuntimeUpdatesProcessor runtimeUpdatesProcessor;
-    private List<HotReplacementSetup> hotReplacement = new ArrayList<>();
-
-    private final Map<Class<?>, Object> liveReloadContext = new ConcurrentHashMap<>();
+    private static volatile CuratedApplication curatedApplication;
+    private Closeable realCloseable;
 
     public DevModeMain(DevModeContext context) {
         this.context = context;
     }
 
     public static void main(String... args) throws Exception {
-        Timing.staticInitStarted();
 
         try (InputStream devModeCp = DevModeMain.class.getClassLoader().getResourceAsStream(DEV_MODE_CONTEXT)) {
             DevModeContext context = (DevModeContext) new ObjectInputStream(new DataInputStream(devModeCp)).readObject();
```

Comment:
`context` doesn't seem to be closed, btw.

Reply:
context is not Closeable

Reply:
Sorry, I meant the instance of DevModeMain.

Reply:
instantiated below at line 45

Reply:
So what about closing the instance of DevModeMain?

Reply:
I have changed it to be closed after the park, but in practice it would have been cleaned up by the normal shutdown hook anyway.